File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F19.P2.java
class Main {
    static long sharedTotal() throws InterruptedException {
        // BEGIN PROBE F19.P2
        var counter = new java.util.concurrent.atomic.AtomicLong(0);
        Runnable bump = () -> { for (int i = 0; i < 1000; i++) counter.incrementAndGet(); };
        var t1 = Thread.startVirtualThread(bump);
        var t2 = Thread.startVirtualThread(bump);
        t1.join();
        t2.join();
        return counter.get();
        // END PROBE F19.P2
    }

    public static void main(String[] args) throws InterruptedException {
        System.out.println(sharedTotal());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F19.P2.kt
fun probe(): Long {
    // BEGIN PROBE F19.P2
    val counter = java.util.concurrent.atomic.AtomicLong(0)
    val t1 = kotlin.concurrent.thread { repeat(1000) { counter.incrementAndGet() } }
    val t2 = kotlin.concurrent.thread { repeat(1000) { counter.incrementAndGet() } }
    t1.join()
    t2.join()
    return counter.get()
    // END PROBE F19.P2
}

fun main() {
    println(probe())
}
